common/scala/src/main/scala/whisk/common/tracing/TraceUtil.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.common.tracing
+
+import java.util
+import java.util.{HashMap, Map}
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.actor.ActorSystem
+import com.github.levkhomich.akka.tracing._
+import whisk.common.{LoggingMarkers, TransactionId}
+
+/**
+  * Trace Utility Class which provides all methods to trace a given service.
+  */
+object TraceUtil{
+
+    private lazy val trace: TracingExtensionImpl = traceVar;
+
+    private var traceVar: TracingExtensionImpl = null;
+
+    private val traceMap : Map[Long, TracedRequest] = new util.HashMap[Long, TracedRequest]()
+
+    var requestCounter: AtomicInteger = new AtomicInteger(1);
+
+    def init (actorSystem: ActorSystem): TracingExtensionImpl = {
+
+      if(traceVar == null)
+        traceVar = TracingExtension.apply(actorSystem);
+
+      return trace;
+    }
+
+  /**
+    * Start a Trace for given service.
+    * @param serviceName Name of Service to be traced
+    * @param transactionId transactionId to which this Trace belongs.
+    * @return TracedRequest which provides details about current service being traced.
+    */
+  def startTrace(serviceName: String, transactionId: TransactionId): Unit = {
```

Comment:
serviceName is usually constant for a microservice deployment, btw.. this should be a global I think